src/components/DeeplinkWrapper/index.website.js

```diff
@@ -21,68 +21,76 @@ const propTypes = {
+        // so that the popup window will only open when we know the short-lived authToken is valid.
+        Session.removeShortLivedAuthToken();
 
-        const expensifyUrl = new URL(CONFIG.EXPENSIFY.NEW_EXPENSIFY_URL);
-        const params = new URLSearchParams();
-        params.set('exitTo', `${window.location.pathname}${window.location.search}${window.location.hash}`);
+        // If the current link is transition from oldDot, we should wait for it to finish,
+        // because we should pass the final path to the desktop app instead of the transition path.
+        if (Str.startsWith(window.location.pathname, Str.normalizeUrl(ROUTES.TRANSITION_BETWEEN_APPS))) {
+            Session.getShortLivedAuthTokenAfterTransition();
+            return;
+        }
         if (!this.props.session.authToken) {
-            const expensifyDeeplinkUrl = `${CONST.DEEPLINK_BASE_URL}${expensifyUrl.host}/transition?${params.toString()}`;
-            this.openRouteInDesktopApp(expensifyDeeplinkUrl);
+            this.openRouteInDesktopApp();
             return;
         }
-        Authentication.getShortLivedAuthToken().then((shortLivedAuthToken) => {
-            params.set('email', this.props.session.email);
-            params.set('shortLivedAuthToken', `${shortLivedAuthToken}`);
-            const expensifyDeeplinkUrl = `${CONST.DEEPLINK_BASE_URL}${expensifyUrl.host}/transition?${params.toString()}`;
-            this.openRouteInDesktopApp(expensifyDeeplinkUrl);
-        }).catch(() => {
-            // If the request is successful, we call the updateAppInstallationCheckStatus before the prompt pops up.
-            // If not, we only need to make sure that the state will be updated.
-            this.updateAppInstallationCheckStatus();
-        });
+        Session.getShortLivedAuthToken();
     }
 
-    updateAppInstallationCheckStatus() {
-        setTimeout(() => {
-            if (!this.focused) {
-                this.setState({appInstallationCheckStatus: CONST.DESKTOP_DEEPLINK_APP_STATE.INSTALLED});
-            } else {
-                this.setState({appInstallationCheckStatus: CONST.DESKTOP_DEEPLINK_APP_STATE.NOT_INSTALLED});
-            }
-        }, 500);
+    componentDidUpdate(prevProps) {
+        if (prevProps.session.shortLivedAuthToken || !this.props.session.shortLivedAuthToken) {
+            return;
+        }
+
+        // Now that there is a shortLivedAuthToken, the route to the desktop app can be opened.
+        this.openRouteInDesktopApp();
     }
 
-    openRouteInDesktopApp(expensifyDeeplinkUrl) {
-        this.updateAppInstallationCheckStatus();
+    openRouteInDesktopApp() {
+        if (this.hasPopupBeenOpenedBefore) {
+            return;
+        }
+
+        // Once the popup has appeared after the page is loaded, it does not need to be displayed again.
+        // e.g. When a user refresh one tab, the popup should not appear again in other tabs.
```

Comment:
I don't understand this comment. Why are tabs relevant in this context? Why would refreshing a tab make popups appear in other tabs?